server/rpc_codec.go

```diff
@@ -96,7 +97,17 @@ func (c *rpcPlusCodec) WriteResponse(r *response, body interface{}, last bool) e
 		Header: map[string]string{},
 	}
 	if err := c.codec.Write(m, body); err != nil {
-		return err
+		c.buf.wbuf.Reset()
+		m = &codec.Message{
+			Method: r.ServiceMethod,
+			Id:     r.Seq,
+			Error:  errors.Wrapf(err, "Unable to encode body").Error(),
+			Type:   codec.Response,
+			Header: map[string]string{},
+		}
+		if err := c.codec.Write(m, struct{}{}); err != nil {
```

Comment:
Do you need to send an empty struct or can you just pass in nil? Nil seems cleaner if possible.